test/k8sT/Policies.go

```diff
@@ -1005,6 +1005,14 @@ var _ = Describe("K8sPolicyTest", func() {
 				} else {
 					ExpectWithOffset(1, err).ToNot(BeNil(), "traffic was redirected to the proxy when it should have not been redirected")
 				}
+
+				if parser == policy.ParserTypeDNS && redirected {
+					By("Checking that Hubble is correctly annotating the DNS names")
+					res := kubectl.HubbleObserve(helpers.CiliumNamespace, ciliumPod,
+						fmt.Sprintf("--last 1 --from-pod %s/%s --to-fqdn %q",
+							namespaceForTest, appPods[helpers.App2], "*.cilium.io"))
+					res.ExpectContainsFilterLine("{.destination_names[0]}", "vagrant-cache.ci.cilium.io")
```

Comment:
Looks like this newly added line is the culprit of the test failures. Tracing a bit of `git blame` it seems like this may rely on #12332, but that PR was already backported to v1.8. @christarazi I think you did the backport of that PR, do you recall anything unusual about the backport of that one?

Anyway, if it's so reliably failing in every test then this shouldn't be a huge effort to reproduce locally & iterate to locate/fix the issue.

Reply:
Sorry I'm not seeing which PR you're referring to. I see the backport PR I've done, but not the upstream PR.

I don't recall any conflicts when doing 1.8 backports. Also, the tests passed the first time in the backport PR (https://github.com/cilium/cilium/pull/12332), so it seems we have a new failure here, rather than in the previous.

Reply:
Oh, I quoted the wrong PR, I meant https://github.com/cilium/cilium/pull/12252 . However if there weren't any conflicts then this won't provide us any simple answers anyway, something else must be missing.

Reply:
I have absolutely no idea what could be wrong here, since the full stdout is:

```
{"time":"2020-07-13T11:21:23.607974208Z","verdict":"FORWARDED","ethernet":{"source":"82:6a:80:39:5f:4b","destination":"5a:7e:bc:be:37:ab"},"IP":{"source":"10.0.0.131","destination":"147.75.38.95","ipVersion":"IPv4"},"l4":{"TCP":{"source_port":56212,"destination_port":80,"flags":{"ACK":true}}},"source":{"ID":277,"identity":25122,"namespace":"202007131320k8spolicytestbasictesttrafficredirectionstoproxytes","labels":["k8s:appSecond=true","k8s:id=app2","k8s:io.cilium.k8s.policy.cluster=default","k8s:io.cilium.k8s.policy.serviceaccount=app2-account","k8s:io.kubernetes.pod.namespace=202007131320k8spolicytestbasictesttrafficredirectionstoproxytes","k8s:zgroup=testapp"],"pod_name":"app2-88b7f8c4b-l784f"},"destination":{"identity":2,"labels":["reserved:world"]},"Type":"L3_L4","node_name":"k8s1","destination_names":["vagrant-cache.ci.cilium.io"],"event_type":{"type":4,"sub_type":3},"traffic_direction":"EGRESS","Summary":"TCP Flags: ACK"}
```

so the `.destination[0]` selector should clearly return `vagrant-cache.ci.cilium.io`.

The output of the failing test also seems weird:

```
[2020-07-07T11:12:48.199Z]           <string>: vagrant-cache.ci.cilium.io
[2020-07-07T11:12:48.199Z]       to be an element of
[2020-07-07T11:12:48.199Z]           <[]interface {} | len:1, cap:1>: [
[2020-07-07T11:12:48.199Z]               [
[2020-07-07T11:12:48.199Z]                   vagrant-cache.ci.cilium.io,
[2020-07-07T11:12:48.199Z]               ],
[2020-07-07T11:12:48.199Z]           ]
```

So... basically... "I could not find `vagrant-cache.ci.cilium.io` in `[ vagrant-cache.ci.cilium.io, ]`." Huh?

I suspect that maybe the backported `ExpectContainsFilterLine` function #12252 is now working properly, but still, I have no idea why it could be. Anyway, I'm going to check it with Delve and see if it leads me anywhere.

Reply:
> So... basically... "I could not find `vagrant-cache.ci.cilium.io` in `[ vagrant-cache.ci.cilium.io, ]`." Huh?

Sounds a bit like a type issue,i.e. the need is of type `string`, but the haystack is of type`[]string`. We had a similar issue before: https://github.com/cilium/cilium/pull/11794 -- but not sure why this would only happen on the v1.8 branch.

Reply:
Ah! So it looks like https://github.com/cilium/cilium/pull/11794 apparently was merged after we branched off to v1.8. In that case, I think https://github.com/cilium/cilium/pull/11794 needs to be backported!

Reply:
Good catch, thanks a lot!